core/src/main/java/org/elasticsearch/action/DocWriteResponse.java

```diff
@@ -128,13 +177,15 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeString(id);
         out.writeZLong(version);
         out.writeBoolean(forcedRefresh);
+        out.writeByte(getOperation().getOp());
     }
 
     static final class Fields {
         static final String _INDEX = "_index";
         static final String _TYPE = "_type";
         static final String _ID = "_id";
         static final String _VERSION = "_version";
+        static final String _OPERATION = "_operation";
```

Comment:
These `Fields` classes have fallen out of favor in the past six months. Rather than add a new constant I'd just add a new string below, similar to what I did when I added `forced_refresh`. If it doesn't bloat the PR too much you can remove the whole `Fields` class and replace with strings at the call sites.